cmake/schemes/url_sha1_boost_ios_library.cmake.in

```diff
@@ -157,6 +161,56 @@ hunter_boost_component_b2_args(
 
 list(APPEND build_opts ${b2_component_opts})
 
+#
+# handle iostreams dependency on zlib and bzip2 from
+# within Hunter
+# TODO: DRY
+#
+string(COMPARE EQUAL "@HUNTER_PACKAGE_COMPONENT@" "iostreams" is_iostreams)
+if(is_iostreams)
+  # by default, use both bzip2 and zlib
+  set(IOSTREAMS_NO_COMPRESSION 0)
+  set(IOSTREAMS_NO_ZLIB  0)
+  set(IOSTREAMS_NO_BZIP2 0)
+  set(IOSTREAMS_ZLIB_SOURCE  ) # empty
+  set(IOSTREAMS_BZIP2_SOURCE ) # empty
+  foreach(arg ${b2_component_opts})
+    string(COMPARE EQUAL "-s" "${arg}" is_s)
+    if(NOT is_s)
+      string(REPLACE "=" ";" nvPair ${arg})
+      list(GET nvPair 0 n)
+      list(GET nvPair 1 v)
+      set("IOSTREAMS_${n}" ${v})
```

Comment:
We don't need to caclulate `IOSTREAMS_*` variables, they are already in defined in this scope. You can use:
```cmake
message("IOSTREAMS_NO_COMPRESSION: ${IOSTREAMS_NO_COMPRESSION}")
```
to check it. The reason of using `@HUNTER_Boost_CMAKE_ARGS@` is that we get **all** of them in one place.